theme.js

```diff
@@ -1,117 +1,84 @@
-    registry.add('js', {
-      name: 'generate-components-for-scss',
-      ext: 'scss',
-      _addon: this,
-      toTree: function(tree) {
-        var destDir = tree.destDir || (tree.inputTree && tree.inputTree.destDir);
-        var isTests = /tests$/.test(destDir);
-        var isSelf = type === 'self';
-        var isParent = type === 'parent';
-
-        var options = {
-          registryType: type,
-          moduleName: this._addon.name,
-          prefix: isSelf ? ('modules/' + this._addon.name) : registry.app.name,
-          baseDir: path.join(this._addon.root, 'addon'),
-          scss: ['styles/components/*.scss']
-        };
-
-        // TODO need a better way of determining app tree from app tests tree
-        switch (true) {
-          case isTests  : return tree;
-          case isSelf   : return jsPreprocessor(tree, options);
-          case isParent : return jsPreprocessor(tree, options);
-          default       : return tree;
-        }
-      }
+    registry.add('htmlbars-ast-plugin', {
+      name: 'transform-ui-table-components',
+      plugin: TransformUITableComponents
     });
   },
 
-  setupTemplatePreprocessing: function(registry) {
-    registry.add('template', {
-      name: 'generate-templates-for-scss',
-      _addon: this,
-      toTree: function(tree) {
-        var options = {
-          prefix: path.join('modules/', this._addon.name),
-          baseDir: path.join(this._addon.root, 'addon'),
-          scss: ['styles/components/*.scss']
-        };
-
-        return templatePreprocessor(tree, options);
-      }
-    });
+  scssPath: 'addon/styles',
+  toScssTree: function() {
+    var scssDir = path.join(this._baseDiskDir(), this.scssPath);
+    var tree = this.treeGenerator(scssDir);
+    return funnel(tree, { include: ['**/*.scss'] });
   },
 
-  setupCssPreprocessing: function(registry) {
-    var self = this;
-    registry.add('css', {
-      name: 'mixin-classes',
-      toTree: function(tree) {
-        return scssPreprocessor(tree, {
-          name: self.name,
-          components: 'components/*.scss'
-        });
-      }
-    });
+  jsPath: 'addon',
+  toJsTree: function() {
+    var jsDir = path.join(this._baseDiskDir(), this.jsPath);
+    var tree = this.treeGenerator(jsDir);
+    return funnel(tree, { include: ['**/*.js'] });
   },
 
-  setupHtmlTransform: function(registry) {
-    registry.add('htmlbars-ast-plugin', {
-      name: 'transform-component-classes',
-      plugin: TransformComponentClasses
-    });
+  hbsPath: 'addon/templates',
+  toHbsTree: function() {
+    var hbsDir = path.join(this._baseDiskDir(), this.hbsPath);
+    var tree = this.treeGenerator(hbsDir);
+    return funnel(tree, { include: ['**/*.hbs'] });
   },
 
-  postprocessTree: function(type, tree) {
-    if (type === 'css') {
-      tree = csso(tree);
-    }
-
-    return tree;
+  _baseDiskDir: function() {
+    return this.nodeModulesPath.replace(/\/node_modules$/, '');
```

Comment:
@ebryn in this case, we need to dir of the theme addon itself, not the consuming project/app. This is so we can generate trees from each theme